tests/test_photoCalib.py

```diff
@@ -650,6 +653,56 @@ def testMakePhotoCalibFromCalibZeroPoint(self):
         self.assertEqual(photoCalib.getInstFluxAtZeroMagnitude(), fluxMag0)
         self.assertFloatsAlmostEqual(photoCalib.getCalibrationErr(), calibrationErr)
 
+    def checkDeprecatedCalibInterface(self, photoCalib):
+        """Check the backwards compatibility Calib interface."""
+
+        # These are now no-ops, but we've added a deprecated interface for backwards compatibility.
+        # TODO DM-18544: test that a deprecation warning is raised.
+        try:
+            lsst.afw.image.PhotoCalib.setThrowOnNegativeFlux(True)
+        except AttributeError:
+            self.fail("Expected setThrowOnNegativeFlux to be available as a no-op.")
+        self.assertFalse(lsst.afw.image.PhotoCalib.getThrowOnNegativeFlux())
+
+        result = photoCalib.getFluxMag0()
+        self.assertEqual(result[0], photoCalib.getInstFluxAtZeroMagnitude())
+        self.assertTrue(np.isnan(result[1]))  # we just return NaN for the second element now
+
+        msg = "PhotoCalib is immutable: create a new `PhotoCalib` with the calibration factor and error"
+        ", or create it like an old Calib object with `makePhotoCalibFromCalibZeroPoint`."
```

Comment:
Is this supposed to concatenate the two lines? I would not expect it to do so without a trailing `\` on the `msg =` line.

Reply:
Multiple adjacent strings are automatically concatenated: https://docs.python.org/3/reference/lexical_analysis.html#string-literal-concatenation